src/models/multi_modal_features.cpp

```diff
@@ -6,13 +6,23 @@
 
 namespace Generators {
 
-MultiModalFeatures::MultiModalFeatures(State& state, MultiModalFeatures::Mode mode, const std::string& name, int64_t num_feature_tokens)
+MultiModalFeatures::MultiModalFeatures(State& state, MultiModalFeatures::Mode mode, const std::string& name,
+                                       int64_t batch_size, int64_t num_feature_tokens)
     : state_{state},
       type_{mode == MultiModalFeatures::Mode::Input
                 ? model_.session_info_->GetInputDataType(name)
                 : model_.session_info_->GetOutputDataType(name)},
       mode_{mode},
       name_{name} {
+  const auto dims = mode_ == MultiModalFeatures::Mode::Input
+                        ? model_.session_info_->GetInputSymbolicShape(name).size()
+                        : model_.session_info_->GetOutputSymbolicShape(name).size();
+
+  // If the model expects 3 dimensions, add a batch dimension
+  if (dims == 3) {
+    shape_.push_back(1);
```

Comment:
Looks like this should be 
```suggestion
    shape_.push_back(batch_size);
```